### PR TITLE
x/isIn

### PR DESCRIPTION
**Pull request by lulldev**
#215

and eager API for isIn

### CODE REVIEW

File: x/isIn.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+ *   isIn(1, new Map([[1, 1], [2, 2], [3, 3]]))
+ * ) // true
+ *
+ * console.log(
+ *   isIn(4, new Map([[1, 1], [2, 2], [3, 3]]))
+ * ) // false
+ * ```
+ */
+const _isIn = function (value, container) {
+  if (!container) {
```
Comment (richytong) [inline x/isIn.js:65]: this check should be container == null

File: x/isIn.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    return false
+  }
+  if (container.constructor == Set) {
+    return container.has(value)
+  }
+  if (container.constructor == Map) {
+    return Array.from(container.values()).includes(value)
+  }
+
+  return includes(value)(container)
```
Comment (richytong) [inline x/isIn.js:75]: the only other data structures that we need are Array and String, both of which should have the includes property function. I would have a check if (typeof container.includes == 'function') {...} and just return false by default

**Review by richytong — CHANGES_REQUESTED** [review]

File: x/isIn.js (hunk 2 above)
Comment (lulldev) [reply x/isIn.js:75]: but what about Object? i use includes for Array/String and Object

Comment (lulldev) [reply x/isIn.js:75]: @richytong

**Comment by FredericHeem** [comment]
Any news on isIn? It would be a great addition.

File: x/isIn.js (hunk 2 above)
Comment (richytong) [reply x/isIn.js:75]: ah that's right, includes handles Array and String and Object

**Review by richytong — APPROVED** [review]
LGTM

File: x/isIn.js (hunk 1 above)
Comment (lulldev) [reply x/isIn.js]: fixed

**Pull request merged**